### ISSUE TITLE
Conflicts between users and guard methods

### ISSUE THREAD

**Issue by DjulLau** [jeremyben/json-server-auth#25]
First of all : thanks a lot for this tool. It's a great starting point in my quest for simple API setups in development or even small personal production projects. It's well written and as a node/express noob it has been a good learning source too !

Obviously one needs to apply guard to "users" resource but as soon as rights are set users middle wares are not applied anymore (no validation for email/password...).

Example (module usage) : 

``` typescript
import jsonServer from 'json-server';
import jsonServerAuth from 'json-server-auth';

const app = jsonServer.create();
const apiRouter = jsonServer.router('db.json');

const rules = jsonServerAuth.rewriter({
    users: 600,
});

app.use('/api', rules);
app.use('/api', jsonServerAuth);
app.use('/api', apiRouter);

// Start
app.listen(4201, '127.0.0.1', function () {
    console.log("Server now listening on 4201");
});
```
If I then with Postman
``` http
POST /api/users
```
with valid jwt token and empty body it creates a new user with no email and no password.

I guess it's because of the re-writer but I'm not sure how to fix this correctly. Maybe by renaming users routes or resources to avoid conflicts between the two ?

**Comment by jeremyben** [comment]
Thank you ! You're right, I'm looking into this, in the meantime you can use `POST /api/register` or `POST /api/signup`.

**Comment by jeremyben** [comment]
Fixed in v2.01 with https://github.com/jeremyben/json-server-auth/commit/a29ec452141f79fc5967538d4a852b3462f2b928

**Comment by DjulLau** [comment]
Thanks for the quick fix! I confirm it's OK for POST but shouldn't it be the same for PUT?

**Comment by jeremyben** [comment]
Good question. Here's my take :
Json-server conforms to REST and uses `PUT` as a way to replace a whole item, not create or simply patch it.

If you look at https://github.com/typicode/json-server#plural-routes (and `/users` is a plural route, not a singular), you can see that `PUT` requires an id parameter (for the item you want to replace), so I think it should be properly guarded.

**Comment by DjulLau** [comment]
Mmm. I'm not sure I see the difference. According to my test with 600 on users a logged user could replace another one, with no email and pwd...

**Comment by jeremyben** [comment]
Right, I see that. This is more complicated than I thought. I will fix that later, but I don't want to overly complicate the app. Remember that this is only meant for development prototypes, to **emulate** most common authorization flows, if you attempt to hack it, you will succeed. Simply go on the happy path for now concerning the `users` collection.

**Comment by DjulLau** [comment]
OK. No problem. I might try to fix it as an exercise. Thanks!

**Comment by jeremyben** [comment]
Fixed in 2.1.0 with https://github.com/jeremyben/json-server-auth/commit/4382ef1a41dfa90734719eb2cc163c355ec0d733
